src/shim/bo.h

```diff
@@ -16,9 +16,22 @@
 #include "drm_local/amdxdna_accel.h"
 #include <string>
 #include <atomic>
+#if defined(__x86_64__) || defined(_M_X64)
+  #include <x86intrin.h>
+#endif
 
 namespace shim_xdna {
 
+  const int LINESIZE = 64;
+  
+  void flush_cache_line(const char *cur) {
+#if defined(__x86_64__) || defined(_M_X64)
+  _mm_clflush(cur);
+#else
+  __builtin___clear_cache((char *)cur, (char *)(cur + LINESIZE));
```

Comment:
This is instruction cache only, correct? I think we are flushing data.  And is it only supported on ARM and MIPS? Also if the cache flush function supports address range, please use it outside the loop. (change clflush_data instead)